pkg/runner/local_runner.go

```diff
@@ -66,14 +68,60 @@ func convertTimeval(t syscall.Timeval) *duration.Duration {
 	}
 }
 
+func (r *localRunner) createDev(inputRootPath string) error {
```

Comment:
There are a couple of things I don't really like about this approach:
- The chroot option is currently conflated with character device creation, while you could argue the two are independent.
- The list of character devices is Linux specific and not flexible. On FreeBSD, /dev/urandom is a symbolic link.
- We don't use any of the `filesystem.Directory` API here, meaning that this code might be prone to symlink following attacks.
- It makes bb_runner partially responsible for setting up the file system, while that is originally what bb_worker is responsible for.
- Relatedly, it makes bb_runner fatter. The idea is that bb_runner remains relatively simple, so people can also write their own ones where needed. Adding the chroot support is all right, but I think that device node creation is unnecessary.

In other words, can't we somehow make this part of bb_worker? Maybe we should extend LocalBuildExecutor and/or BuildDirectory?